projects/plugins/jetpack/modules/publicize/publicize.php

```diff
@@ -419,29 +522,40 @@ private function get_profile_picture( $connection ) {
 	/**
 	 * Whether the user needs to select additional options after connecting
 	 *
-	 * @param string $service_name 'facebook', 'twitter', etc.
-	 * @param object|array The Connection object (WordPress.com) or array (Jetpack)
+	 * @param string       $service_name 'facebook', 'twitter', etc.
+	 * @param object|array $connection The Connection object (WordPress.com) or array (Jetpack).
 	 * @return bool
 	 */
-	function show_options_popup( $service_name, $connection ) {
+	public function show_options_popup( $service_name, $connection ) {
 		$cmeta = $this->get_connection_meta( $connection );
 
-		// always show if no selection has been made for facebook
-		if ( 'facebook' == $service_name && empty( $cmeta['connection_data']['meta']['facebook_profile'] ) && empty( $cmeta['connection_data']['meta']['facebook_page'] ) )
+		// Always show if no selection has been made for Facebook.
+		if ( 'facebook' === $service_name && empty( $cmeta['connection_data']['meta']['facebook_profile'] ) && empty( $cmeta['connection_data']['meta']['facebook_page'] ) ) {
 			return true;
+		}
 
-		// always show if no selection has been made for tumblr
-		if ( 'tumblr' == $service_name && empty ( $cmeta['connection_data']['meta']['tumblr_base_hostname'] ) )
+		// Always show if no selection has been made for Tumblr.
+		if ( 'tumblr' === $service_name && empty( $cmeta['connection_data']['meta']['tumblr_base_hostname'] ) ) {
 			return true;
+		}
 
 		// if we have the specific connection info..
-		if ( isset( $_GET['id'] ) ) {
-			if ( $cmeta['connection_data']['id'] == $_GET['id'] )
+		$id = sanitize_text_field( wp_unslash( $_GET['id'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		if ( $id ) {
+			if ( $cmeta['connection_data']['id'] === $id ) {
 				return true;
+			}
 		} else {
-			// otherwise, just show if this is the completed step / first load
-			if ( !empty( $_GET['action'] ) && 'completed' == $_GET['action'] && !empty( $_GET['service'] ) && $service_name == $_GET['service'] && ! in_array( $_GET['service'], array( 'facebook', 'tumblr' ) ) )
+			// Otherwise, just show if this is the completed step / first load.
+			// phpcs:disable WordPress.Security.NonceVerification.Recommended
+			$action  = sanitize_text_field( wp_unslash( $_GET['action'] ) );
+			$service = sanitize_text_field( wp_unslash( $_GET['service'] ) );
+			// phpcs:enable WordPress.Security.NonceVerification.Recommended
+
+			if ( ! empty( $action ) && 'completed' === $action && ! empty( $service ) && $service_name === $service && ! in_array( $service, array( 'facebook', 'tumblr' ), true ) ) {
```

Comment:
We might need to do the `empty` checks earlier - something like this, maybe?

```suggestion
			$is_completed  = ! empty( $action ) && 'completed' === $_GET['action'] );
			$service       = ! empty( $_GET['service'] ? sanitize_text_field( wp_unslash( $_GET['service'] ) ) : false;
			// phpcs:enable WordPress.Security.NonceVerification.Recommended

			if ( $is_completed && $service_name === $service && ! in_array( $service, array( 'facebook', 'tumblr' ), true ) ) {
```